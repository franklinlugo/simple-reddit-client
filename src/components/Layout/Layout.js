import React from 'react';
import { element } from 'prop-types';
import { Main, Posts, ActivePost } from './Layout.styles';

function Layout({ posts, activePost }) {
  return (
    <Main>
      <Posts>{posts}</Posts>
      <ActivePost>{activePost}</ActivePost>
    </Main>
  );
}
Layout.propTypes = {
  posts: element.isRequired,
  activePost: element.isRequired,
};

export default Layout;
