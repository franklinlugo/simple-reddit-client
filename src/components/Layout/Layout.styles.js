import styled from 'styled-components';

export const Main = styled.main`
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: row;
`;

export const PostsContainer = styled.div`
  width: 500px;
  background-color: ${(props) => props.theme.darkgray};
  border-right: 1px solid ${(props) => props.theme.red};
`;

export const ActivePost = styled.div`
  flex: 1;
`;
