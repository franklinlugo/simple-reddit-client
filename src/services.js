import Snoowrap from 'snoowrap';

const { USER_AGENT, CLIENT_ID, CLIENT_SECRET, REFRESH_TOKEN } = process.env;

const r = new Snoowrap({
  userAgent: USER_AGENT,
  clientId: CLIENT_ID,
  clientSecret: CLIENT_SECRET,
  refreshToken: REFRESH_TOKEN,
});

export async function getTopPost() {
  const topPosts = await r.getTop({ time: 'week', limit: 5 });
  return topPosts;
}
