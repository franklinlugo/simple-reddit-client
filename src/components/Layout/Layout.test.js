import React from 'react';
import { screen } from '@testing-library/react';
import Layout from './Layout';
import { renderWithProvider } from '../../utils/tests/renderWithProvider';
import { getTopPost as getTopPostMock } from '../../services';
import { postsList } from '../../utils/tests/fixtures';

jest.mock('../../services');

afterAll(() => {
  jest.clearAllMocks();
});

test('Layout should render', async () => {
  getTopPostMock.mockResolvedValueOnce(postsList);
  const { asFragment } = renderWithProvider(<Layout />);
  await screen.findByText('111 title');
  expect(asFragment()).toMatchSnapshot();
});
