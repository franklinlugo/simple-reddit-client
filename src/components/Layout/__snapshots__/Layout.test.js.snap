// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 {
  width: 100%;
  padding: 1rem;
  border-bottom: 1px solid;
}

.c3 {
  display: grid;
  grid-template-columns: repeat(2,min-content);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 1rem;
  column-gap: 1rem;
}

.c4 {
  margin: 0;
  padding: 0;
  line-height: 1;
  white-space: nowrap;
}

.c5 {
  margin: 1rem 0;
  color: 60;
  white-space: nowrap;
}

.c6 {
  display: grid;
  grid-template-columns: 100px 1fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 1rem;
  column-gap: 1rem;
  margin-top: 1rem;
}

.c7 {
  width: 100px;
  height: 100px;
}

.c8 {
  margin: 0;
  padding: 0;
  line-height: 1;
}

.c9 {
  display: grid;
  grid-template-columns: repeat(2,min-content);
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 1rem;
}

.c10 {
  display: grid;
  grid-template-columns: repeat(2,min-content);
  white-space: nowrap;
  border: 1px solid;
  cursor: pointer;
}

.c11 {
  white-space: nowrap;
}

.c1 {
  width: 500px;
  border-right: 1px solid;
}

.c12 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: fixed;
  top: 0;
  width: calc(100% - 500px);
  height: 100%;
  margin-left: 500px;
}

<main
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h3
            class="c4"
          >
            111 author
          </h3>
          <h5
            class="c5"
          >
            5/5/1970, 9:00:00 PM
          </h5>
        </div>
        <div
          class="c6"
        >
          <img
            class="c7"
            src="https://image1111.com"
          />
          <p
            class="c8"
          >
            111 title
          </p>
        </div>
        <div
          class="c9"
        >
          <button
            class="c10"
          >
            Dismiss Post
          </button>
          <span
            class="c11"
          >
             0 Comments
          </span>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h3
            class="c4"
          >
            222 author
          </h3>
          <h5
            class="c5"
          >
            5/5/1970, 9:00:00 PM
          </h5>
        </div>
        <div
          class="c6"
        >
          <img
            class="c7"
            src="https://image222.com"
          />
          <p
            class="c8"
          >
            222 title
          </p>
        </div>
        <div
          class="c9"
        >
          <button
            class="c10"
          >
            Dismiss Post
          </button>
          <span
            class="c11"
          >
             0 Comments
          </span>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h3
            class="c4"
          >
            333 author
          </h3>
          <h5
            class="c5"
          >
            5/5/1970, 9:00:00 PM
          </h5>
        </div>
        <div
          class="c6"
        >
          <img
            class="c7"
            src="https://image333.com"
          />
          <p
            class="c8"
          >
            333 title
          </p>
        </div>
        <div
          class="c9"
        >
          <button
            class="c10"
          >
            Dismiss Post
          </button>
          <span
            class="c11"
          >
             0 Comments
          </span>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h3
            class="c4"
          >
            444 author
          </h3>
          <h5
            class="c5"
          >
            5/5/1970, 9:00:00 PM
          </h5>
        </div>
        <div
          class="c6"
        >
          <img
            class="c7"
            src="https://image444.com"
          />
          <p
            class="c8"
          >
            444 title
          </p>
        </div>
        <div
          class="c9"
        >
          <button
            class="c10"
          >
            Dismiss Post
          </button>
          <span
            class="c11"
          >
             0 Comments
          </span>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h3
            class="c4"
          >
            555 author
          </h3>
          <h5
            class="c5"
          >
            5/5/1970, 9:00:00 PM
          </h5>
        </div>
        <div
          class="c6"
        >
          <img
            class="c7"
            src="https://image555.com"
          />
          <p
            class="c8"
          >
            555 title
          </p>
        </div>
        <div
          class="c9"
        >
          <button
            class="c10"
          >
            Dismiss Post
          </button>
          <span
            class="c11"
          >
             0 Comments
          </span>
        </div>
      </div>
    </div>
    <div
      class="c12"
    />
  </main>
</DocumentFragment>
`;
